Quadriga/src/main/java/edu/asu/spring/quadriga/service/publicwebsite/impl/ProjectStats.java

```diff
@@ -0,0 +1,198 @@
+
+        List<IProjectCollaborator> projectCollaboratorList = projectCollaboratorManager
+                .getProjectCollaborators(projectId);
+
+        if (projectCollaboratorList == null)
+            return null;
+
+        Map<String, IUserStats> mapUserWorkspace = new HashMap<String, IUserStats>();
+        // loop through each collaborator
+
+        for (IProjectCollaborator projectCollaborator : projectCollaboratorList) {
+            if (projectCollaborator.getCollaborator() != null) {
+                String username = projectCollaborator.getCollaborator()
+                        .getUserObj().getUserName();
+                UserStats userStats = (UserStats) setUserWorkspaceStats(
+                        projectId, username);
+                mapUserWorkspace.put(username, userStats);
+            }
+        }
+
+        String username = projectDeepMapper.getProjectDetails(projectId)
+                .getOwner().getUserName();
+        if (username == null)
+            return mapUserWorkspace;
+
+        UserStats userStats = (UserStats) setUserWorkspaceStats(projectId,
+                username);
+        mapUserWorkspace.put(username, userStats);
+
+        return mapUserWorkspace;
+    }
+
+    public IUserStats setUserWorkspaceStats(String projectId, String username)
+            throws QuadrigaStorageException {
+        List<IWorkSpace> workspaceList = listWSDAO.listWorkspace(projectId,
+                username);
+        UserStats userStats;
+        if (workspaceList == null) {
+            userStats = new UserStats(username, 0, 0);
+        } else {
+            userStats = new UserStats(username, workspaceList.size(), 0);
+        }
+        return userStats;
+
+    }
+
+    @Override
+    public Map<String, IUserStats> getCountofNetwork(String projectid)
+            throws QuadrigaStorageException {
+        List<INetwork> networksList = networkManager
+                .getNetworksInProject(projectid);
+
+        if (networksList == null)
+            return null;
+
+        Map<String, IUserStats> mapUserNetworks = new HashMap<String, IUserStats>();
+
+        for (INetwork network : networksList) {
+            if (network.getCreator() != null) {
+                String username = network.getCreator().getUserName();
+                IUserStats userStats = mapUserNetworks.get(username);
+                if (userStats == null) {
+                    userStats = new UserStats(username, 0, 1);
+                } else {
+                    userStats.incrementNetworkCount();
+                }
+                mapUserNetworks.put(username, userStats);
+            }
+        }
+        return mapUserNetworks;
+    }
+
+    @Override
+    public List<IUserStats> getUserStats(String projectId)
+            throws QuadrigaStorageException {
+
+        Map<String, IUserStats> mapWorkspaceStats = getCountofWorkspace(projectId);
+        Map<String, IUserStats> mapNetworkStats = getCountofNetwork(projectId);

```

Comment:
Now, I'm getting confused. Are you calling getCountofNetwork and getCountofWorkspace anywhere outside of this class? If not and if they are not intended to, then these methods should be private. Also, in this case create the map once and hand it over to your private methods to be filled so that you can get rid of the extra looping below. If you intend for other classes to be able to call getCountofNetwork and getCountofWorkspace, and have getUserStats as a helper method to get both stats at once, then I would revise your UserStats class, so that you do not have one value always be 0, but either two separate classes for networks and workspaces (possible combined through a third one that this class returns), or just return maps in getCountofNetwork and getCountofWorkspace. Does this make sense?